apps/omg_watcher/lib/exit_processor.ex

```diff
@@ -246,9 +254,25 @@ defmodule OMG.Watcher.ExitProcessor do
   """
   def handle_call(:check_validity, _from, state) do
     # NOTE: future of using `ExitProcessor.Request` struct not certain, see that module for details
-    {chain_status, events} =
+    {request, state} =
       %ExitProcessor.Request{}
       |> run_status_gets()
+      # To find if IFE was included, see first if its inputs were spent.
+      |> Core.determine_ife_input_utxos_existence_to_get(state)
+      |> run_ife_input_utxo_existance()
+      # Next, check by what transactions they were spent.
+      |> Core.determine_ife_spends_to_get(state)
+      |> run_ife_spend_getting()
+      # Find tx bodies.
+      |> Core.determine_ife_blocks_to_get()
+      |> run_ife_block_getting()
+      # Compare found txes with ife.tx.
+      # If equal, persist information about position.
+      |> Core.find_ifes_in_blocks(state)
+
+    {chain_status, events} =
+      request
+      # main loop
```

Comment:
not a `loop` really. Maybe just remove that comment? in this pipeline, the function's names are more explanatory

(we could probably do better in both pipelines, but let's not rush this refactoring, there's many other things that we could refactor here first)

(2 occurences)

Reply:
different thing - one refactor that we probably could do here (shame on me, that I didn't already), is to put the common part of this bit (the "main loop") into `defp prepare_common_request_data(request, state)` (or some better name).

Reply:
done - DRYed into private function